crates/uv-resolver/src/resolver/resolver_markers.rs

```diff
@@ -20,6 +22,19 @@ pub enum ResolverMarkers {
 impl ResolverMarkers {
     /// Set the resolver to perform a resolution for a specific environment.
     pub fn specific_environment(markers: MarkerEnvironment) -> Self {
+        // The resolver operates with release-only semantics for Python versions. So if the user's
+        // environment specifies a pre-release version, we need to strip it.
+        let python_full_version = markers.python_full_version().only_release();
```

Comment:
Can we guarantee pre-releases are stripped in `MarkerEnvironment` instead?

Reply:
I don't think they should be... I think `MarkerEnvironment` should be capable of representing the true markers. Otherwise, we'll misreport (for example) when we send telemetry up to PyPI about the user's Python version.

Reply:
(I did consider that.)

Reply:
(Do you disagree?)

Reply:
Hmm okay that makes sense. I guess `ResolverMarkers` is the correct place to put this then.

Reply:
I'll a distinct type to enforce this.